server/src/infra/repositories/person.repository.ts

```diff
@@ -96,14 +96,32 @@ export class PersonRepository implements IPersonRepository {
     return this.personRepository.findOne({ where: { id: personId } });
   }
 
-  getByName(userId: string, personName: string): Promise<PersonEntity[]> {
-    return this.personRepository
+  getByName(userId: string, personName: string, withHidden: boolean | undefined): Promise<PersonEntity[]> {
```

Comment:
This implementation has a lot of overlap with `getAllForUser`, can you see if it is possible to use the same repository method for both implementations? They both basically use the same sorting criteria, query, etc. The parts that different would turn into search options (limit, etc). I think something like `searchPeople(userId: string, options: SearchOptions): Promise<PersonEntity[]>;` would make sense instead of this additional variation.